src/hostcalls.rs

```diff
@@ -318,6 +318,15 @@ pub fn set_map_value(map_type: MapType, key: &str, value: Option<&str>) -> Resul
     }
 }
 
+pub fn remove_map_value(map_type: MapType, key: &str) -> Result<(), Status> {
+    unsafe {
+        match proxy_remove_header_map_value(map_type, key.as_ptr(), key.len()) {
+            Status::Ok => Ok(()),
+            status => panic!("unexpected status: {}", status as u32),
+        }
+    }
+}
+
```

Comment:
Style: Could you move it above, so that it's not in-between `set_map_value` and `set_map_value_bytes`?

Ideally, could you group it with the `extern C` prototype, so that it's:
```
extern "C" {
    fn proxy_remove_header_map_value(
    [...]
}

pub fn remove_map_value(map_type: MapType, key: &str) -> Result<(), Status> {
[...]

extern "C" {
    fn proxy_replace_header_map_value(
    [...]
}

pub fn set_map_value(map_type: MapType, key: &str, value: Option<&str>) -> Result<(), Status> {
[...]
```

to match the existing style in this file?

Reply:
Nit: The `remove` group should be before `set` group, since the latter is using `proxy_remove_header_map_value`, which is now used before declaration.